modules/social_features/social_out_of_office/social_out_of_office.module

```diff
@@ -0,0 +1,256 @@
+        $profile_storage = \Drupal::entityTypeManager()->getStorage('profile');
+
+        /** @var \Drupal\profile\Entity\ProfileInterface $profile */
+        $alt_contact_person = $profile_storage->loadByUser($alt_contact_person_user, SocialOutOfOfficeHelperInterface::PROFILE_TYPE);
+
+        if ($alt_contact_person instanceof ProfileInterface) {
+          /** @var \Drupal\social_profile\SocialProfileNameService $profile_name_service */
+          $profile_name_service = \Drupal::service('social_profile.name_service');
+
+          // Get generated profile name.
+          $alt_contact_person_name = $profile_name_service->getProfileName($alt_contact_person);
+          // Prepare contact person URL.
+          $alt_contact_person_url = Url::fromRoute('social_user.stream', ['user' => $alt_contact_person->id()]);
+
+          // Add extra variables with necessary data.
+          $variables['profile_alt_contact_person_name'] = $alt_contact_person_name;
+          $variables['profile_alt_contact_person_url'] = $alt_contact_person_url;
+        }
+      }
+    }
+  }
+}
+
+/**
+ * Implements hook_preprocess_user().
+ */
+function social_out_of_office_preprocess_user(array &$variables): void {
+  // Show status in private messaging thread if someone is out of office.
+  if ($variables['elements']['#view_mode'] === 'private_message_author') {
+    $current_user = \Drupal::currentUser()->getAccount();
+    $user = $variables['elements']['#user'];
+
+    if ($current_user->id() !== $user->id()) {
+      /** @var \Drupal\profile\ProfileStorageInterface $profile_storage */
+      $profile_storage = \Drupal::entityTypeManager()->getStorage('profile');
+
+      /** @var \Drupal\profile\Entity\Profile $profile */
+      $profile = $profile_storage->loadByUser($user, SocialOutOfOfficeHelperInterface::PROFILE_TYPE);
+
+      if ($profile instanceof ProfileInterface) {
+        /** @var \Drupal\social_out_of_office\SocialOutOfOfficeHelperInterface $ooo_helper */
+        $ooo_helper = \Drupal::service('social_out_of_office.helper');
+
+        // Check if user is out of office.
+        if ($ooo_helper->isUserOutOfOffice($profile)) {
+          // Retrieve out of office dates.
+          $dates = $ooo_helper->getOutOfOfficeDates($profile);
+
+          $suffix = ' ' . t('is out of office from @start_date to @end_date', [
+            '@start_date' => $dates['start'],
+            '@end_date' => $dates['end'],
+          ]);
+          $suffix .= $variables['content']['linked_username']['#suffix'];
+          $variables['content']['linked_username']['#suffix'] = $suffix;
+        }
+      }
+
+      // Add OoO cache tag.
+      $variables['#cache']['tags'][] = 'ooo_dates:uid:' . $user->id();
+    }
+  }
+}
+
+/**
+ * Implements hook_entity_presave().
+ */
+function social_out_of_office_entity_presave(EntityInterface $entity): void {
+  // Show message about users that out of office if someone mentioned them in
+  // post or comment.
+  if (
```

Comment:
Related to removing dependencies: no post/comment entities? OK, this code won't break, see https://3v4l.org/vFMvN